kore/test/Test/Kore/Step/Simplification/Integration.hs

```diff
@@ -557,12 +553,26 @@ test_simplificationIntegration =
         let expected = OrPattern.fromPatterns
                 [ Conditional
                     { term = mkTop Mock.otherSort
-                    , predicate = makeInPredicate Mock.otherSort
-                        (mkNu iz (Mock.builtinInt 595))
-                        (mkAnd
-                            (Mock.tdivInt mkTop_ mkTop_)
-                            (mkAnd
-                                (mkCeil_
+                    , predicate =
+                        makeAndPredicate
+                            (makeAndPredicate
+                                (makeCeilPredicate Mock.otherSort
+                                    (mkAnd
+                                        (Mock.tdivInt mkTop_ mkTop_)
+                                        (mkNu iz (Mock.builtinInt 595))
+                                    )
+                                )
+                                (makeAndPredicate
+                                    (makeCeilPredicate Mock.otherSort
+                                        (Mock.tdivInt mkTop_ mkTop_)
+                                    )
+                                    (makeCeilPredicate Mock.otherSort
+                                        (mkNu iz (Mock.builtinInt 595))
+                                    )
+                                )
```

Comment:
This is redundant. The first condition ensures that each of these subterms is defined. Do you know where this is coming from?

Reply:
I was under the impression that `ceil(\and(a, b))` will also generate `\and(\ceil(a), \ceil(b))` if it can't infer that `a` and `b` are defined: https://github.com/kframework/kore/blob/dceca511119b17383c6f07e98914a30e4815de32/kore/src/Kore/Step/Simplification/Ceil.hs#L350 Am I misunderstanding this? I checked for this example and the terms appear in the `childCeils`.

Reply:
OK, it looks like this is doing what is intended.